File: src/main/java/ordilov/lissn/playlist/domain/youtube/Snippet.java
package ordilov.lissn.playlist.domain.youtube;

import java.util.List;
import lombok.Getter;

@Getter
public class Snippet {

  @Getter
  public static class PlaylistItem {

    private String title;
    private String description;
    private String publishedAt;
    private Thumbnails thumbnails;
    private String channelTitle;
    private String channelId;
    private String playlistId;
    private String position;
    private ResourceId resourceId;
    private String videoOwnerChannelTitle;
    private String videoOwnerChannelId;

    @Getter
    public static class ResourceId {

      private String videoId;
      private String kind;
    }
  }

  @Getter
  public static class Video {

    private String title;
    private String channelId;
    private String categoryId;
    private List<String> tags;
    private String publishedAt;
    private String description;
    private String channelTitle;
    private Thumbnails thumbnails;
    private String liveBroadcastContent;
    private String defaultAudioLanguage;
    private Localized localized;
  }

  @Getter
  private static class Localized {
    private String title;
    private String description;
  }
}

File: src/main/java/ordilov/lissn/playlist/domain/youtube/Snippet.kt
package ordilov.lissn.playlist.domain.youtube

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

data class PlaylistItem(
    val title: String,
    val description: String,
    val publishedAt: String,
    val thumbnails: Thumbnails,
    val channelTitle: String,
    val channelId: String,
    val playlistId: String,
    val position: String,
    val resourceId: ResourceId,
    val videoOwnerChannelTitle: String,
    val videoOwnerChannelId: String,
)

data class ResourceId(
    val videoId: String,
    val kind: String,
)

@JsonIgnoreProperties(ignoreUnknown = true)
data class Video(
    val title: String,
    val channelId: String,
    val categoryId: String,
    val publishedAt: String,
    val description: String,
    val channelTitle: String,
    val thumbnails: Thumbnails,
    val liveBroadcastContent: String,
    val defaultAudioLanguage: String,
    val localized: Localized,
    val tags: List<String>
)

data class Localized(
    val title: String,
    val description: String
)